controller/backup_controller.go

```diff
@@ -279,29 +287,97 @@ func (bc *BackupController) reconcile(backupName string) (err error) {
 	}()
 
 	// Perform backup snapshot to remote backup target
-	if backup.Spec.SnapshotName != "" && backup.Status.State == "" {
-		// Initialize a backup target client
-		backupTargetClient, err := getBackupTargetClient(bc.ds, backupTarget)
-		if err != nil {
-			log.WithError(err).Error("Error init backup target client")
-			return nil // Ignore error to prevent enqueue
-		}
+	if backup.Spec.SnapshotName != "" {
+		switch backup.Status.State {
+		case types.BackupStateNew:
+			// Initialize a backup target client
+			backupTargetClient, err := getBackupTargetClient(bc.ds, backupTarget)
+			if err != nil {
+				log.WithError(err).Error("Error init backup target client")
+				return nil // Ignore error to prevent enqueue
+			}
 
-		// Initialize a engine client
-		engine, err := bc.ds.GetVolumeCurrentEngine(backupVolumeName)
-		if err != nil {
-			return err
-		}
-		engineCollection := &engineapi.EngineCollection{}
-		engineClient, err := GetClientForEngine(engine, engineCollection, engine.Status.CurrentImage)
-		if err != nil {
-			return err
-		}
+			// Initialize a engine client
+			engine, err := bc.ds.GetVolumeCurrentEngine(backupVolumeName)
+			if err != nil {
+				return err
+			}
+			engineCollection := &engineapi.EngineCollection{}
+			engineClient, err := GetClientForEngine(engine, engineCollection, engine.Status.CurrentImage)
+			if err != nil {
+				return err
+			}
 
-		if err := bc.backupCreation(log, engineClient, backupTargetClient.URL, backupTargetClient.Credential, backup); err != nil {
-			return err
+			if err := bc.backupCreation(log, engineClient, backupTargetClient.URL, backupTargetClient.Credential, backup); err != nil {
+				return err
+			}
+
+			// Monitor snapshot backup progress
+			bc.enableBackupMonitor(backup, engineClient)
+			return nil
+		case types.BackupStateInProgress:
+			bm := bc.hasMonitor(backup.Name)
+			if bm == nil || bm.engineBackupStatus == nil {
+				return nil
+			}
+
+			backup.Status.Progress = bm.engineBackupStatus.Progress
+			backup.Status.URL = bm.engineBackupStatus.BackupURL
+			backup.Status.Error = bm.engineBackupStatus.Error
+			backup.Status.SnapshotName = bm.engineBackupStatus.SnapshotName
+			backup.Status.ReplicaAddress = bm.engineBackupStatus.ReplicaAddress
+			// https://github.com/longhorn/longhorn-engine/blob/9da3616/pkg/replica/backup.go#L20-L22
+			switch bm.engineBackupStatus.State {
+			case "in_progress":
+				backup.Status.State = types.BackupStateInProgress
+			case "complete":
+				backup.Status.State = types.BackupStateCompleted
+			case "error":
+				backup.Status.State = types.BackupStateError
+			}
+
```

Comment:
NIT: consider moving this mapping/conversion function (engineBackupStatus -> backup.Status) into a util function.